src/app/ApplicationTemplate.Tests/Presentation/ChuckNorrisSearchPageViewModelShould.cs

```diff
@@ -99,46 +78,40 @@ private void CheckForSearchMethodWhenQuotesAreLoading_SpecialConfiguration(IHost
 						.Setup(m => m.Search(It.IsAny<CancellationToken>(), It.IsAny<string>()))
 						.ReturnsAsync(new ChuckNorrisQuote[]
 						{
-							new ChuckNorrisQuote(new ChuckNorrisData.Builder().WithId("0"), false),
-							new ChuckNorrisQuote(new ChuckNorrisData.Builder().WithId("1203"), false)
+							new ChuckNorrisQuote(new ChuckNorrisData.Builder().WithId("0").WithValue("Something something dog"), false),
+							new ChuckNorrisQuote(new ChuckNorrisData.Builder().WithId("1203").WithValue("Dog something"), false)
 						});
 
-					mock
-						.Setup(m => m.GetFavorites(It.IsAny<CancellationToken>()))
-						.ReturnsAsync(new SourceList<ChuckNorrisQuote>().AsObservableList());
+					MockingGetFavorites(mock);
 				});
 			});
 		}
 
-		/// <summary>
-		/// Verify that the Search method from ChuckNorrisService is:
-		/// - Exactly called once if the search term is valid.
-		/// - Never called if not.
-		/// </summary>
-		/// <param name="expectedCallCount">Number of times the Search method must be called.</param>
-		/// <param name="searchTerm">The term used for the search.</param>
-		/// <param name="chuckNorrisServiceMock">The <see cref="IChuckNorrisService"/> used for the search (mocked).</param>
-		/// <returns><see cref="Task"/>.</returns>
-		[Theory]
-		[InlineAutoData("test")]
-		public async Task CheckForSearchMethodWhenQuotesAreLoading(string searchTerm)
+		[Fact]
+		public async Task CheckForSearchMethod_WhenQuotesAreLoading()
 		{
 			// Arrange
-			var chuckNorrisServiceMock = new Mock<IChuckNorrisService>();
-			ConfigurationSetup(CheckForSearchMethodWhenQuotesAreLoading_SpecialConfiguration);
-
-			var searchPageViewModel = new ChuckNorrisSearchPageViewModel();
+			ConfigurationSetup(CheckForSearchMethod_WhenQuotesAreLoading_SpecialConfiguration);
+			var viewModel = new ChuckNorrisSearchPageViewModel();
 
 			// Act
-			searchPageViewModel.SearchTerm = searchTerm;
-			var quotes = await searchPageViewModel.Quotes.Load(DefaultCancellationToken);
+			viewModel.SearchTerm = "dog";
+			var quotes = await viewModel.Quotes.Load(DefaultCancellationToken);
 
 			// Assert
-			quotes.Length
-				.Should().Be(2);
+			using (new AssertionScope())
+			{
+				quotes.Should()
+					.NotBeNull()
+					.And
+					.NotBeEmpty();
+
+				quotes.All(q => q.Quote.Value.ToUpperInvariant().Contains("DOG"))
+					.Should().BeTrue("All quotes found searching for search term 'dog' should contain 'dog' ");
+			}
 		}
 
-		private void CheckForMultipleSearchMethodCall_SpecialConfiguration(IHostBuilder host)
+		private void GiveDifferentResults_ForMultipleSearchMethodCall_SpecialConfiguration(IHostBuilder host)
```

Comment:
I think we prefer "Return" to "Give" ?

Reply:
Should the verbs be conjugated? Like **Returns** instead of **Return**?

Reply:
Oh. I guess I was confused because this one starts directly with the verb, not the *ClassNameShould*.

Reply:
I will write some doc. ClassNameShould is the name of the class, nothing to do with the naming of the tests